course_info/tests/jasmine/controllers/PeopleControllerSpec.js

```diff
@@ -128,19 +160,540 @@ describe('Unit testing PeopleController', function() {
+        });
+
+        describe('clearSearchResults', function() {
+            it('should empty any search results', function() {
+                scope.searchResults = [{}];  // contents don't matter
+                scope.clearSearchResults();
+                expect(scope.searchResults).toEqual([]);
+            });
+        });
+
+        describe('closeAlert', function() {
+            it('should work when an alert is present', function() {
+                scope.testAlerts = [{}];  // contents don't matter
+                scope.closeAlert('testAlerts', 0);
+                expect(scope.testAlerts).toEqual([]);
+            });
+            
+            it('should not blow up when no alert is present', function() {
+                scope.testAlerts = [];
+                scope.closeAlert('testAlerts', 0);
+                expect(scope.testAlerts).toEqual([]);
+            });
+
+            it('should only remove the index requested', function() {
+                scope.testAlerts = [1, 2, 3];
+                scope.closeAlert('testAlerts', 1);
+                expect(scope.testAlerts).toEqual([1,3]);
+            });
+        });
+
+        describe('compareRoles', function() {
+            it('should sort as expected', function() {
+                var roles = [
+                    {active: 0, prime_role_indicator: ''},
+                    {active: 1, prime_role_indicator: ''},
+                    {active: 0, prime_role_indicator: 'N'},
+                    {active: 1, prime_role_indicator: 'N'},
+                    {active: 0, prime_role_indicator: 'Y'},
+                    {active: 1, prime_role_indicator: 'Y'},
+                ];
+                var afterSorting = [
+                    {active: 1, prime_role_indicator: 'Y'},
+                    {active: 1, prime_role_indicator: 'N'},
+                    {active: 1, prime_role_indicator: ''},
+                    {active: 0, prime_role_indicator: 'Y'},
+                    {active: 0, prime_role_indicator: 'N'},
+                    {active: 0, prime_role_indicator: ''},
+                ];
+
+                roles.sort(scope.compareRoles);
+                expect(roles).toEqual(afterSorting);
+            });
+        });
+
+        describe('disableAddUserButton', function() {
+            // the state of the add user button depends on the contents of
+            // the search field, on whether a lookup has returned multiple
+            // results, and on whether any of those results have been selected
+            it('should disable if a search is in progress', function() {
+                scope.searchInProgress = true;
+                expect(scope.disableAddUserButton()).toBe(true);
+            });
+            it('should enable if the search had multiple results and one is selected',
+               function() {
+                   scope.searchResults = [{}]; // contents don't matter, only length
```

Comment:
I see in PeopleController.js that `$scope.searchResults` is set iff there are multiples; it's a little confusing though that 'multiple results' as described in the `it()` clause is equivalent to a single-item array, since there's not a clear indication by the scope variable name that it will only be populated if there are multiple results. Could these maybe be `[{}, {}]` just to be a little more life-like?